docs/development/javascript/how-to-create-and-use-single-file-components-vuejs/index.md

```diff
@@ -0,0 +1,1271 @@
+
+    {{< note >}}
+Local registration is a valuable architectural feature for reusable components within big projects.
+{{< /note >}}
+
+-   The `HelloWorld` component is used within the `App` component's template on line 4.
+
+## Building your First Single File Components
+
+Now that we’ve covered the basic structure of the project created by Vue CLI, let's build our own components on top of that. As in [Building and Using VueJS Components](/docs/development/javascript/how-to-build-and-use-vuejs-components/), we will again be building a rating application, but this time it will be a little more sophisticated.
+
+This is what your rating app will look like:
+
+![Rating App - Finished Product](rating-app-finished-product.png "Rating App - Finished Product")
+
+This is how it will behave:
+
+-   Clicking on a star on the left side will register a vote for that star.
+
+-   The left side will interactively change when a user hovers over the stars.
+
+-   It will allow the user to rate only once on each visit to the page. If the page is refreshed, or if it is visited again later, the user can vote again.
+
+-   It will keep score of votes between page visits in the browser's local storage.
+
+Here's how the app's template will look in protocode; you do *not* need to copy and paste this:
+
+{{< file "" html >}}
+<div id="app">
+    <div class="inner">
+        <div class="ratingContainer">
+            <span class="bigRating"></span>
+            <div class="rating-stars">
+                <Star weight="1"></Star>
+                <Star weight="2"></Star>
+                <Star weight="3"></Star>
+                <Star weight="4"></Star>
+                <Star weight="5"></Star>
+            </div>
+        </div>
+        <Summary></Summary>
+    </div>
+</div>
+{{< /file >}}
+
+We’ll make each star a separate component (named `Star`), and we'll also create a `Summary` component which will hold the summary of the votes.
+
+### App.vue
+
+To start, replace the content of your `App.vue` with this snippet:
+
+{{< file "src/App.vue" html >}}
+<template>
+    <div id="app">
+        <div class="inner">
+            <div class="ratingContainer">
+                <span class="bigRating" v-html="bigRating"></span>
+                <div>
+                    <Star
+                        v-for="index in 5"
+                        v-bind:key="index"
+                        v-bind:weight="index"
+                        v-bind:enabled="enabled"
+                        v-bind:currentRating="currentRating"
+                    ></Star>
+                </div>
+            </div>
+            <Summary v-bind:ratings="ratings"></Summary>
+        </div>
+    </div>
+</template>
+
+<script>
+import Star from "./components/Star";
```

Comment:
Does Star need a .vue extension here? Same in the following line.

Reply:
it actually works without the extension [because of a webpack option](https://webpack.js.org/configuration/resolve/#resolveenforceextension), but I'm gonna add them in anyway to be clearer

Reply:
fixed